lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -477,6 +477,37 @@ public void forceDeleteTask(String taskId)
     deleteTask(taskId, true);
   }
 
+  @Override
+  public Task selectAndClaim(TaskQuery taskQuery)
+      throws TaskNotFoundException, NotAuthorizedException, InvalidStateException,
+          InvalidOwnerException {
+
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("entry to selectAndClaim(taskQuery = {})", taskQuery);
+    }
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      ((TaskQueryImpl) taskQuery).selectAndClaimEquals(true);
+
+      TaskSummary taskSummary = taskQuery.single();
+
+      if (taskSummary == null) {
+        throw new SystemException(
+            "No tasks matched the specified filter and sorting options,"
+                + " task query returned nothing!");
+      }
```

Comment:
why SystemException? According to JavaDoc I expected a "TaskNotFoundExeption"

Reply:
The TaskNotFoundException from JavaDoc is inherited from the claim() call. While I would agree that in case of a null return from the query a TaskNotFoundException seems logical, it needs an id, which we usually won't have here, we only got a query (assuming taskIdIn is not always set).